appscale/tools/remote_helper.py

```diff
@@ -122,7 +121,24 @@ def start_head_node(cls, options, my_id, node_layout):
       options.verbose)
 
     if options.infrastructure:
-      instance_id, public_ip, private_ip = cls.spawn_node_in_cloud(options)
+      agent = InfrastructureAgentFactory.create_agent(options.infrastructure)
+
+      params = agent.get_params_from_args(options)
+      public_ips, private_ips, instance_ids = agent.describe_instances(params)
+      try:
+        login_ip = LocalState.get_login_host(options.keyname)
+      except (IOError, BadConfigurationException):
+        # We don't have the location file, we need to start instances.
```

Comment:
If we don't have the location file the instances are effectively lost. We could in theory find them running under the proper keyname,  but we wouldn't know the role of each one of them.